spokestack/wakeword/tflite.py

```diff
@@ -101,67 +111,70 @@ def __call__(self, context: SpeechContext, frame) -> None:
             self.reset()
 
     def _sample(self, context: SpeechContext, frame) -> None:
-        # normalize incoming audio to (-1.0, 1.0)
+        # convert the PCM-16 audio to float32 in (-1.0, 1.0)
         frame = frame.astype(np.float32) / (2 ** 15 - 1)
-        # ensure range does not exceed (-1.0, 1.0)
         frame = np.clip(frame, -1.0, 1.0)
-        # pull out a single value from the frame
+
+        # pull out a single value from the frame and apply pre-emphasis
+        # with the previous sample then cache the previous sample
+        # to be use in the next iteration
         prev_sample = frame[-1]
-        # apply pre-emphasis with the previous sample
         frame -= self.pre_emphasis * np.append(self._prev_sample, frame[:-1])
-        # cache the previous sample to be use in the next iteration
         self._prev_sample = prev_sample
 
+        # fill the sample window to analyze speech containing samples
+        # after each window fill the buffer advances by the hop length
+        # to produce an overlapping window
         for sample in frame:
             self.sample_window.write(sample)
-            # fill the sample window
             if self.sample_window.is_full:
-                # run analyze if speech detected in the vad
                 if context.is_speech:
                     self._analyze(context)
-                # prepare the sample window for the next iteration
                 self.sample_window.rewind().seek(self.hop_length)
 
     def _analyze(self, context: SpeechContext) -> None:
-        # read the full contents of the sample window
+        # read the full contents of the sample window to calculate a single frame
+        # of the STFT by applying the DFT to a real-valued input and
+        # polarize the output by computing the absolute value
```

Comment:
I wouldn't use the term polarize here - something like "take the magnitude of the complex DFT"